pkg/scheduler/objects/queue.go

```diff
@@ -1028,6 +1028,12 @@ func (sq *Queue) IncAllocatedResource(alloc *resources.Resource, nodeReported bo
 func (sq *Queue) allocatedResFits(alloc *resources.Resource) bool {
 	sq.RLock()
 	defer sq.RUnlock()
+	// on the root we want to reject a new allocation if it asks for resources not registered
+	// so do not use the "undefined" flag, also handles pruned max for root
+	if sq.isRoot() {
+		return sq.maxResource.FitIn(resources.AddOnlyExisting(alloc, sq.allocatedResource))
+	}
+	// any other queue undefined is always good
```

Comment:
Can we have a single line check for all queues? Is Internal FitIn method private? Can it be made public?

Reply:
Root queue often needs special casing, as we have here. I'm fine with the code as-is.